lib/game.rb

```diff
@@ -0,0 +1,124 @@
+class Game
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6],
+  ]
+
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+      @board = board
+      @player_1 = player_1
+      @player_2 = player_2
+    end
+
+  def won?
+    no_win = true
+    WIN_COMBINATIONS.each do |win_array|
+      win = []
+      win_array.select do |position|
+        if @board.taken?(position + 1)
+          win << position
+          if win.length == 3 && win_array == win && (@board.cells[win[0]] == @board.cells[win[1]] && @board.cells[win[0]] == @board.cells[win[2]])
+            no_win = false
+            return win_array
+          end
+        end
+      end
+    end
+    if no_win
+     false
+    end
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def draw?
+    !won? && @board.full?
+  end
+
+  def winner
+    won? ? @board.position((won?[0] + 1).to_s) : nil
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.